analytics/views/support.py

```diff
@@ -68,11 +69,15 @@ def get_plan_name(plan_type: int) -> str:
 
 
 def get_confirmations(
-    types: List[int], object_ids: List[int], hostname: Optional[str] = None
+    types: List[int],
+    confirmation_queryset_or_ids: Union[
+        Iterable[int], QuerySet[MultiuseInvite], QuerySet[PreregistrationUser]
+    ],
+    hostname: Optional[str] = None,
 ) -> List[Dict[str, Any]]:
     lowest_datetime = timezone_now() - timedelta(days=30)
     confirmations = Confirmation.objects.filter(
-        type__in=types, object_id__in=object_ids, date_sent__gte=lowest_datetime
+        type__in=types, object_id__in=confirmation_queryset_or_ids, date_sent__gte=lowest_datetime
```

Comment:
So my thinking had been that a cleaner approach would be to just change the two other callers to get IDs rather than objects -- they don't appear to use the database queries they're doing for anything other than calling this function.

Reply:
As Anders [mentioned](https://github.com/zulip/zulip/pull/22292#discussion_r916182559), passing a `QuerySet` into the `in` query makes only one (nested) SQL query ([querysets/#in](https://docs.djangoproject.com/en/4.0/ref/models/querysets/#in)). Forcing those querysets into id lists will result in more queries being executed.

However, I suppose that performance is not that critical for this code path, so it makes sense to sacrifice it for clarity. What do you think?

Reply:
Certainly that sort of performance consideration isn't important here. And even if we're passing a QuerySet, it's likely more readable to pass a ValuesQuerySet of integers...